test/metabase/pulse/pulse_integration_test.clj

```diff
@@ -43,11 +47,41 @@
                                                                   :base_type     :type/Float
                                                                   :semantic_type :type/Percentage
                                                                   :field_ref     [:field "Tax Rate" {:base-type :type/Float}]}]}
-                     Card {question-card-id :id} {:name          "Query based on model"
+                    Card {~question-card-id :id} {:name          "Query based on model"
                                                   :dataset_query {:type     :query
                                                                   :database (mt/id)
-                                                                  :query    {:source-table (format "card__%s" model-card-id)}}}
-                     Dashboard {dash-id :id} {:name "just dash"}
+                                                                  :query    {:source-table (format "card__%s" ~'model-card-id)}}}]
+       ~@body)))
+
+(defn- run-pulse-and-return-last-data-columns
```

Comment:
This is super handy. 
I can see adding a couple helpers here with a similar pattern, but with `return last-data-row`, and perhaps some other little things like that.

In the case of checking column formatting on exports, for example, you can check a couple formatting settings in one fell swoop by pulling a row and ensuring each column has the right format... 

Anyway, I'm trying to say I like this ns with the macro and the helper fn :)